tests/unit/device.ts

```diff
@@ -277,12 +277,15 @@ describe('Device', function() {
           await device.connect();
           sinon.assert.calledOnce(stub);
           assert(!!device['_activeCall']);
+          assert(!!device['_makeCallPromise']);
         });
 
         it('should reject if inputDevicePromise rejects', async () => {
           device.audio!._getInputDevicePromise = () => new Promise((resolve, reject) => reject());
           await assert.rejects(() => device.connect());
           assert(!device['_activeCall']);
+          assert.notDeepEqual(device['_makeCallPromise'], null)
```

Comment:
This brings up a bug. I'm glad we discussed and caught it. If the makeCallPromise rejects, any future setIntputDevice will fail. For example, if you do the following

```ts
this.device.audio.setInputDevice('wrong-id');
this.device.connect();
this.device.audio.setInputDevice('default');
```

The 3rd line fails although the ID is correct. This happens because we're not clearing the makeCallPromise. I think to resolve this, we need to make sure the makeCallPromise internal reference is cleared after it is resolved or rejected. I'm not sure off top of my head where to do that, can you please see what we should change in this implementation to make sure this bug doesn't happen?

Reply:
Revised, let me know what you think